File: .tmp_py4013/batch_12_dump/3400_maximum_number_of_matching_indices_after_right_shifts.js
// LeetCode 3400 - Maximum Number of Matching Indices After Right Shifts
// https://leetcode.com/problems/maximum-number-of-matching-indices-after-right-shifts/

var maximumMatchingIndices = function(nums1, nums2) {
    const n = nums1.length;
    let ans = 0;
    for (let shift = 0; shift < n; shift++) {
        let cnt = 0;
        for (let i = 0; i < n; i++) {
            if (nums1[(i - shift + n) % n] === nums2[i]) cnt++;
        }
        if (cnt > ans) ans = cnt;
    }
    return ans;
};

File: .tmp_rs4013/rust_check17/3400_maximum_number_of_matching_indices_after_right_shifts.rs
struct Solution;
// LeetCode 3400 - Maximum Number of Matching Indices After Right Shifts
// https://leetcode.com/problems/maximum-number-of-matching-indices-after-right-shifts/

impl Solution {
    pub fn maximum_matching_indices(nums1: Vec<i32>, nums2: Vec<i32>) -> i32 {
        let n = nums1.len();
        let mut ans = 0;
        for shift in 0..n {
            let mut cnt = 0;
            for i in 0..n {
                if nums1[(i + n - shift) % n] == nums2[i] {
                    cnt += 1;
                }
            }
            if cnt > ans {
                ans = cnt;
            }
        }
        ans
    }
}

fn main() {}
